docs/preview/features/message-pumps/service-bus.md

```diff
@@ -302,6 +303,58 @@ string transactionId = correlationInfo.TransactionId;
 string operationId = correlationInfo.OperationId;
 ```
 
+## Automatic Azure Key Vault credentials rotation
+
+The additional library `Arcus.Messaging.Pumps.ServiceBus.KeyRotation` provides an extension on the message pump to restart the pump automatically when the credentials of the pump stored in Azure Key Vault are changed.
+This feature allows more reliable restarting instead of relying on authentication exceptions that may be throwed during the lifetime of the message pump.
+
+## How does this work?
+
+A background job will be run in the background which is in fact a message pump on a Service Bus Topic.
+The Azure Key Vault where the connection string for your target message pump is located should send `SecretNewVersionCreated` events to this Service Bus Topic.
+
+That way, when the background job receives a new Key Vault event, it will know that your target message pump should be restarted.
+
+### Installation
+
+This features requires to install our NuGet package:
+
+```shell
+PM > Install-Package Arcus.Messaging.Pumps.ServiceBus.KeyRotation
+```
+
+### Usage
+
+When the package is installed, you'll be able to use the extension in your application:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    // You should have a unique Job ID to identity the message pump so the automatic process knows which pump to restart.
+    string jobId = Guid.NewGuid().ToString();
+
+    string secretName = hostContext.Configuration["ARCUS_KEYVAULT_CONNECTIONSTRINGSECRETNAME"];
+    services.AddServiceBusQueueMessagePump(secretName, options => options.JobId = jobId)
+            .WithServiceBusMessageHandler<OrdersAzureServiceBusMessageHandler, Order>();
+
+    // This extension will be available to you once you installed the package.
+    services.WithAutoRestartOnRotatedCredentials(
+        jobId: jobId, 
+        subscriptionNamePrefix: "TestSub", 
+        
+        // The secret key where the Azure Service Bus Topic connection string is located that the background job will use to receive the Azure Key vault events.
+        serviceBusTopicConnectionStringSecretKey: "ARCUS_KEYVAULT_SECRETNEWVERSIONCREATED_CONNECTIONSTRING",
+        
+        // The secret key where the Azure Service Bus connection string is located that your target message pump uses.
+        // This secret key name will be used to check if the received Azure Key Vault event is from this secret or not.
+        targetConnectionStringKey: secretName,
```

Comment:
```suggestion
        messagePumpConnectionStringKey: secretName,
```

But ideally we pull this in automatically since we know it already as part of https://github.com/arcus-azure/arcus.messaging/pull/107/files#diff-1f44b85c1f99fedb43b7e3a702a84b81d8ca6b38533d1a607ecec94295e9cc09R337

Reply:
Hmm, I think the link isn't pointing to what you want it to point.

Reply:
I should have used a commit but basically you know the secret already because it was passed to the pump

Reply:
That's not entirely correct, as it's used directly inline with `await secretProvider.GetRawSecretAsync(secretName)` the secret name is not saved (yet) in the settings. We could do that off course.

Reply:
And also, when the user used the `Func<ISecretProvider, Task<string>>` we don't know what the secret is...

Reply:
Ok, fair enough.